### PR TITLE
Hotfix/retry sample tracking

### PR DESCRIPTION
**Pull request by wow-such-code**

### CODE REVIEW

**Comment by wow-such-code** [comment]
[Retry sample tracking update once or twice in all relevant ETL scripts](https://trello.com/c/IFdEEa9J/436-retry-sample-tracking-update-once-or-twice-in-all-relevant-etl-scripts)

File: drop-boxes/register-IdXML-dropbox/register-IdXML-dropbox.py [hunk 1]
```diff
@@ -103,4 +104,15 @@ def process(transaction):
         transaction.moveFile(incomingPath, dataSet)
 
         #sample tracking section
-        SAMPLE_TRACKER.updateSampleLocationToCurrentLocation(identifier)
+        wait_seconds = 1
+        try_count = 3
+        for i in range(try_count):
```
Comment (sven1103) [inline drop-boxes/register-IdXML-dropbox/register-IdXML-dropbox.py]: maybe some more descriptive variable name, like `attempt` and `max_attempts`

File: drop-boxes/register-archived-ms-raw/register-raw.py [hunk 2]
```diff
@@ -104,4 +105,16 @@ def process(transaction):
...
+                try:
+                        SAMPLE_TRACKER.updateSampleLocationToCurrentLocation(parentCode)
+                        break
+                except:
+                        print "Updating location for sample "+parentCode+" failed on attempt "+str(attempt+1)
+                        if attempt < max_attempts -1:
+                                time.sleep(wait_seconds)
+                                continue
+                        else:
+                                raise
```
Comment (sven1103) [inline drop-boxes/register-archived-ms-raw/register-raw.py]:
```suggestion
                                raise

```

File: drop-boxes/register-qpcr-dropbox/register-qpcr-dropbox.py [hunk 3]
```diff
@@ -103,4 +104,16 @@ def process(transaction):
...
+                try:
+                        SAMPLE_TRACKER.updateSampleLocationToCurrentLocation(parentCode)
+                        break
+                except:
+                        print "Updating location for sample "+parentCode+" failed on attempt "+str(attempt+1)
+                        if attempt < max_attempts -1:
+                                time.sleep(wait_seconds)
+                                continue
+                        else:
+                                raise
```
Comment (sven1103) [inline drop-boxes/register-qpcr-dropbox/register-qpcr-dropbox.py]:
```suggestion
                                raise
                                
```

File: drop-boxes/register-vcf-dropbox/register-vcf.py [hunk 4]
```diff
@@ -163,4 +164,16 @@ def process(transaction):
...
+                try:
+                        SAMPLE_TRACKER.updateSampleLocationToCurrentLocation(parentCode)
+                        break
+                except:
+                        print "Updating location for sample "+parentCode+" failed on attempt "+str(attempt+1)
+                        if attempt < max_attempts -1:
+                                time.sleep(wait_seconds)
+                                continue
+                        else:
+                                raise
```
Comment (sven1103) [inline drop-boxes/register-vcf-dropbox/register-vcf.py]:
```suggestion
                                raise
                                
```

**Review by sven1103 — DISMISSED** [review]
nice !

**Pull request merged**